core/src/confidence.rs

```diff
@@ -1,113 +1,59 @@
-use crate::consensus::StakeLockout;
+use crate::result::{Error, Result};
 use crate::service::Service;
-use std::collections::{HashMap, HashSet};
+use solana_runtime::bank::Bank;
+use solana_vote_api::vote_state::VoteState;
+use solana_vote_api::vote_state::MAX_LOCKOUT_HISTORY;
+use std::collections::HashMap;
 use std::sync::atomic::{AtomicBool, Ordering};
-use std::sync::mpsc::{channel, Receiver, Sender};
+use std::sync::mpsc::{channel, Receiver, RecvTimeoutError, Sender};
 use std::sync::{Arc, RwLock};
 use std::thread::{self, Builder, JoinHandle};
+use std::time::Duration;
 
-#[derive(Debug, Default, PartialEq)]
-pub struct Confidence {
-    fork_stakes: u64,
-    total_stake: u64,
-    lockouts: u64,
-    stake_weighted_lockouts: u128,
+#[derive(Debug, Default, Eq, PartialEq)]
+pub struct BankConfidence {
+    confidence: [u64; MAX_LOCKOUT_HISTORY],
 }
 
-impl Confidence {
-    pub fn new(fork_stakes: u64, total_stake: u64, lockouts: u64) -> Self {
-        Self {
-            fork_stakes,
-            total_stake,
-            lockouts,
-            stake_weighted_lockouts: 0,
-        }
+impl BankConfidence {
+    pub fn increase_confirmation_stake(&mut self, confirmation_count: usize, stake: u64) {
+        assert!(confirmation_count > 0 && confirmation_count <= MAX_LOCKOUT_HISTORY);
+        self.confidence[confirmation_count - 1] += stake;
     }
-    pub fn new_with_stake_weighted(
-        fork_stakes: u64,
-        total_stake: u64,
-        lockouts: u64,
-        stake_weighted_lockouts: u128,
-    ) -> Self {
-        Self {
-            fork_stakes,
-            total_stake,
-            lockouts,
-            stake_weighted_lockouts,
-        }
+
+    pub fn get_confirmation_stake(&mut self, confirmation_count: usize) -> u64 {
+        assert!(confirmation_count > 0 && confirmation_count <= MAX_LOCKOUT_HISTORY);
+        self.confidence[confirmation_count - 1]
     }
 }
 
-#[derive(Default, PartialEq)]
+#[derive(Default)]
 pub struct ForkConfidenceCache {
-    confidence: HashMap<u64, Confidence>,
+    bank_confidence: HashMap<u64, BankConfidence>,
+    _total_stake: u64,
```

Comment:
What's your plan with this field?

Reply:
hehe plan to use it as a denominator when we need to get percentages to respond to validators